generate/generate_file_test.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/generate"
+)
+
+func TestGenerateFile(t *testing.T) {
+	checkGenFiles := func(t *testing.T, got string, want string) {
```

Comment:
as `t` is a pointer, passing it as a parameter doesn't help much, then using it as a closure variable saves a few characters when calling checkGenFiles()... it's a minor change then up to you.

Reply:
Can't do that in this case, subtests are created and if a subtest calls Fail on a parent test *testing.T Go panics since that is not allowed (basically I need the subtests to pass their own testing instance pointers).

Reply:
ah they're not the same pointer, each subtest has its own *testing.T. gotcha.